docs/blog/posts/generator.md

```diff
@@ -0,0 +1,189 @@
+    {"product_id": 1, "product_name": "Apple MacBook Air (2023) - Latest model, high performance, portable"},
+    {"product_id": 2, "product_name": "Sony WH-1000XM4 Wireless Headphones - Noise-canceling, long battery life"},
+    {"product_id": 3, "product_name": "Apple Watch Series 7 - Advanced fitness tracking, seamless integration with Apple ecosystem"},
+    {"product_id": 4, "product_name": "Kindle Oasis - Premium e-reader with adjustable warm light"},
+    {"product_id": 5, "product_name": "AllBirds Wool Runners - Comfortable, eco-friendly sneakers"},
+    {"product_id": 6, "product_name": "Manduka PRO Yoga Mat - High-quality, durable, eco-friendly"},
+    {"product_id": 7, "product_name": "Bench Hooded Jacket - Stylish, durable, suitable for outdoor activities"},
+    {"product_id": 8, "product_name": "GoPro HERO9 Black - 5K video, waterproof, for action photography"},
+    {"product_id": 9, "product_name": "Nespresso Vertuo Next Coffee Machine - Quality coffee, easy to use, compact design"},
+    {"product_id": 10, "product_name": "Project Hail Mary by Andy Weir - Latest sci-fi book from a renowned author"}
+]
+```
+
+Let's not define our models for structured extraction. Note: instructor will conveniently let us use `Iterable` to model an iterable of our class. In this case, once we define our product recommendation model, we can slap on `Iterable` to define what we ultimately want - a (ranked) list of product recommendations. 
+
+
+```python
+import instructor
+from openai import OpenAI
+from typing import Iterable
+from pydantic import BaseModel
+
+client = instructor.patch(OpenAI(), mode=instructor.function_calls.Mode.JSON)
+
+class ProductRecommendation(BaseModel):
+    product_id: str
+    product_name: str
+    relevance_score: float
+
+Recommendations = Iterable[ProductRecommendation]
+```
+Now let's use our instructor patch. Since we don't want to wait for all the tokens to finish, will set stream to `True` and process each product recommendation as it comes in:
+```python
+
+prompt = f"Based on the following user profile:\n{profile_data}\nRank the following products from most relevant to least relevant:\n" + '\n'.join(f"{product['product_id']} {product['product_name']}" for product in products)
+
+
+recommendations_stream = client.chat.completions.create(
+    model="gpt-3.5-turbo-1106",
+    temperature=0.1,
+    response_model=Recommendations,
+    stream=True,
+    messages=[
+        {"role": "system", "content": "Generate product recommendations based on the customer profile. Return in order of highest recommended first."},
+        {"role": "user", "content": prompt}
+    ]
+)
+```
+
+`recommendations_stream` is a generator! It yields the extracted products as it's processing the stream in real-time. We can simply iterate over this:
+
+```python
+for recommendation in recommendations_stream:
+    assert isinstance(recommendation, ProductRecommendation)
+    print(f"Recommended Product: {recommendation.product_name}")
```

Comment:
It would be nice to conclude by highlighting that your web application now displays results faster. Even a 100ms improvement can lead to a 1% increase in revenue. These improvements are significant. We can even utilize the previous benchmarking code to demonstrate the increased speed in obtaining the first result.